### PR TITLE
Add option to add authentication to Jaeger and custom tags

### PR DESCRIPTION
**Pull request by aledbf**
## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/13381
Fixes https://github.com/gitpod-io/gitpod/issues/13385

## How to test
- Configure the installer setting 
```
tracing:
  secretName: XXXXX
```

and verify the env variables`JAEGER_USER` and `JAEGER_PASSWORD` are configured.

## Release Notes

```release-note
Add option to add authentication to Jaeger and configure default tags with `metadata.shortName` and `metadata.region`
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Review by jenting — APPROVED** [review]

**Review by jeanp413 — APPROVED** [review]

File: install/installer/pkg/common/common.go [hunk 1]
```diff
@@ -154,6 +154,41 @@ func tracingEnv(context *RenderContext, tracing *experimental.Tracing) (res []co
...
+		})
+
+		res = append(res, corev1.EnvVar{
+			Name: "JAEGER_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: *context.Config.Observability.Tracing.SecretName},
+				Key:                  "JAEGER_PASSWORD",
+			}},
+		})
+	}
```
Comment (easyCZ) [inline install/installer/pkg/common/common.go:173]: Why do we expose these as ENV variables? In general, mounting the secret as a volume is preferred for credentials and here we could do that if the SecretName is specified.

File: install/installer/pkg/common/common.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func DefaultEnv(cfg *config.Config) []corev1.EnvVar {
 	)
 }
 
-func WorkspaceTracingEnv(context *RenderContext) (res []corev1.EnvVar) {
+func WorkspaceTracingEnv(context *RenderContext, component string) (res []corev1.EnvVar) {
```
Comment (easyCZ) [inline install/installer/pkg/common/common.go:115]: Do we need this indirection here if we're actually keeping them the same? Could we instead consolidate on `func TracingEnv(...)`. If the need for these to actually be different arises, we can refactor.

File: install/installer/pkg/common/common.go (hunk 1 above)
Comment (aledbf) [reply install/installer/pkg/common/common.go:173]: That's how the [library is configured](https://github.com/jaegertracing/jaeger-client-go/blob/8d8e8fcfd04de42b8482476abac6a902fca47c18/README.md#environment-variables).

https://github.com/gitpod-io/gitpod/blob/45b0b45ec09ab90e09f24c620b56eef94ca52673/components/common-go/tracing/tracing.go#L42

File: install/installer/pkg/common/common.go (hunk 2 above)
Comment (aledbf) [reply install/installer/pkg/common/common.go:115]: I don't know the reason for this structure.
ping @MrSimonEmms

File: install/installer/pkg/common/common.go (hunk 1 above)
Comment (easyCZ) [reply install/installer/pkg/common/common.go:173]: We could construct that manually, from the information we mount - https://pkg.go.dev/github.com/uber/jaeger-client-go/config#Configuration

I'll raise an issue as a follow-up (which will likely never get done) as I don't want to block your change but we do abuse env variables a fair bit.

**Review by easyCZ — APPROVED** [review]

File: install/installer/pkg/common/common.go (hunk 2 above)
Comment (MrSimonEmms) [reply install/installer/pkg/common/common.go:115]: Nothing to do with me I'm afraid. This was added by @andrew-farries around 5 months ago - https://github.com/gitpod-io/gitpod/commit/1119e5578bfe502ae99b90708649ee19523c8cc0

**Comment by atduarte** [comment]
@aledbf updated release notes as it didn't seem to be useful for the end Gitpod user. Let me know if this change is wrong.

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by aledbf** [comment]
@atduarte it depends on who is the "Gitpod user". If you are installing Gitpod, this is an important thing to know.

**Comment by atduarte** [comment]
@aledbf I understand the confusion. Release Notes are automatically added to the gitpod.io changelog ([October's](https://github.com/gitpod-io/website/pull/2883), [September's](https://www.gitpod.io/changelog/september-gitpod-release-2022)) and therefore should focus only on the end-user (i.e. the developer persona). Some additional internal information can be found here: [1](https://gitpod.slack.com/archives/C01KGM9AW4W/p1654677957766019), [2](https://www.notion.so/gitpod/Release-Notes-513a74fdd23b4cb1b3b3aefb1d34a3e0).

**Comment by loujaybee** [comment]
> @atduarte it depends on who is the "Gitpod user". If you are installing Gitpod, this is an important thing to know - @aledbf 

You are correct that we should treat the installer as a first-class citizen for the Self-Hosted context, however see below for reasons on why we should omit for now 👇 

> Release Notes are automatically added to the gitpod.io changelog (https://github.com/gitpod-io/website/pull/2883, [September's](https://www.gitpod.io/changelog/september-gitpod-release-2022)) and therefore should focus only on the end-user - @atduarte 

Currently, yes. That's also my understanding. @lucasvaltl can confirm, but I believe Self-Hosted are cherry-picking PR's and manually curating the release notes right now ([see past example](https://www.gitpod.io/changelog/september-self-hosted-release)) and omitting **Release Notes**. No heuristic exists (that I'm aware of) to distinguish Self-Hosted / installer facing pull requests from SaaS or developer persona pull requests, that heuristic needs creating and agree'ing on, then we can easily distinguish between the two.

**Pull request merged**